### PR TITLE
Ask notification permission in the app

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/1210

## Proposed Changes

Show notification dialog :
  - If the user hasn't enabled them
  - The app open count is > 3
  - The dialog has never been showed

### CODE REVIEW

File: ooniprobe/View/TabsViewController.m [hunk 1]
```diff
@@ -61,12 +63,17 @@ -(void)viewDidAppear:(BOOL)animated{
...
+             && ![SettingsUtility isNotificationEnabled]){
+        [MessageUtility notificationAlertinView:self];
+        [SettingsUtility incrementAppOpenCount];
+    }
 }
 
 -(void)setModalValue:(BOOL)value key:(NSString*)key popupName:(NSString*)popupName{
     [[NSUserDefaults standardUserDefaults] setObject:@"ok" forKey:popupName];
     [[NSUserDefaults standardUserDefaults] setBool:value forKey:key];
-    [[NSUserDefaults standardUserDefaults] synchronize];
```
Comment (lorenzoPrimi) [inline ooniprobe/View/TabsViewController.m:69]: this is deprecated and not needed anymore

File: ooniprobe/Utility/SettingsUtility.mm [hunk 2]
```diff
@@ -168,4 +169,20 @@ + (NSString*)getOrGenerateUUID4{
...
 }
 
++ (void)incrementAppOpenCount{
+    if ([self getAppOpenCount] > NOTIFICATION_POPUP_COUNT) return;
+    [[NSUserDefaults standardUserDefaults] setInteger:[self getAppOpenCount]+1 forKey:NOTIFICATION_POPUP];
+}
+
++ (long)getAppOpenCount{
+    long count = [[NSUserDefaults standardUserDefaults] integerForKey:NOTIFICATION_POPUP];
+    if(count < 0) count = 0;
```
Comment (bassosimone) [inline ooniprobe/Utility/SettingsUtility.mm:179]: 👏

File: ooniprobe/View/TabsViewController.m (hunk 1 above)
Comment (bassosimone) [reply ooniprobe/View/TabsViewController.m:69]: 🙏

File: ooniprobe/Model/Database/Measurement.mm [hunk 3]
```diff
@@ -32,6 +32,11 @@ + (NSMutableOrderedSet*)getReportsUploaded {
     return reportIds;
 }
 
++ (long)doneMeasurementsCount {
+    SRKQuery *query = [[Measurement query] where:BASIC_QUERY];
+    return [query count];
+}
+
```
Comment (bassosimone) [inline ooniprobe/Model/Database/Measurement.mm:39]: This code is not used. Can you please remove it from the diff?

**Review by bassosimone — CHANGES_REQUESTED** [review]

**Review by bassosimone — APPROVED** [review]

**Pull request merged**